handler/src/main/java/io/netty/handler/ssl/JdkAlpnApplicationProtocolNegotiator.java

```diff
@@ -117,4 +107,28 @@ public JdkAlpnApplicationProtocolNegotiator(ProtocolSelectorFactory selectorFact
             ProtocolSelectionListenerFactory listenerFactory, String... protocols) {
         super(ALPN_WRAPPER, selectorFactory, listenerFactory, protocols);
     }
+
+    private static final class FailureWrapper implements SslEngineWrapperFactory {
+        @Override
+        public SSLEngine wrapSslEngine(SSLEngine engine, JdkApplicationProtocolNegotiator applicationNegotiator,
+                                       boolean isServer) {
+            throw new RuntimeException("ALPN unsupported. Is your classpatch configured correctly?"
+                    + " See http://www.eclipse.org/jetty/documentation/current/alpn-chapter.html#alpn-starting");
+        }
+    }
+
+    private static final class AlpnWrapper implements SslEngineWrapperFactory {
+        @Override
+        public SSLEngine wrapSslEngine(SSLEngine engine, JdkApplicationProtocolNegotiator applicationNegotiator,
+                                       boolean isServer) {
+            if (ConscryptAlpnSslEngine.isEngineSupported(engine)) {
```

Comment:
yes my preference would be to make this configurable (default to conscrypt) because its based upon class path resolution.